src/Workleap.OpenApi.MSBuild/Spectral/SpectralRulesetManager.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Microsoft.Build.Framework;
+using YamlDotNet.RepresentationModel;
+
+namespace Workleap.OpenApi.MSBuild.Spectral;
+
+internal sealed class SpectralRulesetManager
+{
+    private const string WorkleapRulesetVersion = "0.9.0";
+    private const string WorkleapRulesetDownloadUrlFormat = "https://raw.githubusercontent.com/gsoft-inc/wl-api-guidelines/{0}/.spectral.{1}.yaml";
+    
+    private readonly ILoggerWrapper _loggerWrapper;
+    private readonly IHttpClientWrapper _httpClientWrapper;
+    private readonly string _spectralProfile;
+    private readonly string _spectralRulesetPath;
+
+    public SpectralRulesetManager(
+        ILoggerWrapper loggerWrapper, 
+        IHttpClientWrapper httpClientWrapper,
+        string spectralProfile,
+        string? customSpectralFilePath)
+    {
+        this._loggerWrapper = loggerWrapper;
+        this._httpClientWrapper = httpClientWrapper;
+
+        this._spectralProfile = spectralProfile;
+
+        this._spectralRulesetPath = customSpectralFilePath ?? GetProfileRulesetUrl(this._spectralProfile);
+    }
+
+    private static string GetProfileRulesetUrl(string spectralProfile)
+    {
+        return string.Format(WorkleapRulesetDownloadUrlFormat, WorkleapRulesetVersion, spectralProfile);
+    }
+
+    public async Task<string> GetLocalSpectralRulesetFile(CancellationToken cancellationToken)
+    {
+        var rulesetPath = this._spectralRulesetPath;
+        
+        // For remote ruleset we download the file for optimization and reduce spectral flakiness
+        if (!IsLocalFile(rulesetPath))
```

Comment:
nit: it feels confusing that we have to put negation in front of IsLocalFile() method and it is a private method only used here. I think naming can be improved

Reply:
Updated!